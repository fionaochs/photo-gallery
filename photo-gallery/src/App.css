@import url('https://fonts.googleapis.com/css?family=Heebo|Oswald');

body {
    font-family: 'Heebo', "Segoe UI", Candara, "Bitstream Vera Sans", "DejaVu Sans", "Bitstream Vera Sans", "Trebuchet MS", Verdana, "Verdana Ref", sans-serif;
    background: rgb(65, 34, 205);
    margin: 0;
    line-height: 1.5em;
}

h1 {
    font-size: 2em;
}

h1, h2 {
    font-family: 'Oswald';
    font-weight: bold;
}
* {
    box-sizing: border-box;
}

header, main {
    padding: 10px;
}

header {
    display: flex;
    align-items: center;
    justify-content: space-around;
    background: rgb(144, 175, 242);
    border-bottom: 1px solid teal;
    margin-top: 20px;
}

h1 {
    margin: 0 20px;
    flex-grow: 1;
}

.options {
    padding: 10px 0;
}

section {
    margin: 20px auto;
    width: 90%;
    min-width: 650px;
}


.creatures {
    display: flex;
    flex-wrap: wrap;
}

.horned-types{
  /* display: flex; */
  align-items: center;
}
.creatures li {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    text-align: center;
    height: 350px;
    width: 200px;
    min-width: 200px;
    border: 1px solid #aaa;
    background-color: rgb(144, 175, 242);
    margin-top: 0px;
    margin-right: 10px;
    margin-bottom: 10px;
    margin-left: 0px;
    padding: 10px;
}

.creatures li h2 {
    font-size: 1.5em;
}

.creatures li .creature-type {
    font-size: .8em;
    color: rgb(62, 57, 126);
}

.creatures li .year {
    font-size: .8em;
}

.image-container {
    height: 50%;
}

.image-container img {
    height: 100%;
    object-fit: contain;
    width: 80%;
}

.creature-type-filter {
    height: 48px;
    width: 150px;
    font-size: 1.2em;
}