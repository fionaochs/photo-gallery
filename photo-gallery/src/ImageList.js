import React, { Component } from 'react'

export default class CreatureList extends Component {
    render () {
        return <section className="options">
            
        <section className="list-section">
            <ul className="creatures">{this.props.creaturesNodes}</ul>
        </section>

        </section>
    }
}
