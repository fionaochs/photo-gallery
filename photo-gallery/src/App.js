import React, { Component } from 'react';
import images from './data.js';
import './App.css';
import Creature from './ImageItem.js';
import Header from './Header.js';
import ImageList from './ImageList.js';


class App extends Component {
    state = {
      selected: null
    };
    handleChange = e => {
      this.setState({selected: e.target.value});
    }
    render () {

      const { selected } = this.state;
      const creaturesNodes = images
      .filter(Creature => {
        if (!selected) return true;
        return Creature.keyword === selected;
      })
      .map(hornedCreature => 
        <Creature
        creature={hornedCreature} 
        />)
        
        
        return (
          <div className="App">
      <Header></Header>
      <select className="horned-types" onChange={this.handleChange}>
                <option value="" defaultValue>
                    All Horned Types
                </option>
                <option value="narwhal">Narwhal</option>
                <option value="rhino">Rhino</option>
                <option value="unicorn">Unicorn</option>
                <option value="unilego">Unilego</option>
                <option value="triceratops">Triceratops</option>
                <option value="markhor">Markhor</option>
                <option value="mouflon">Mouflon</option>
                <option value="addax">Addax</option>
                <option value="chameleon">Chameleon</option>
                <option value="lizard">Lizard</option>
                <option value="dragon">Dragon</option>
            </select>
      <ImageList creaturesNodes={creaturesNodes}></ImageList>
      </div>
      );
    }      
}
export default App;

