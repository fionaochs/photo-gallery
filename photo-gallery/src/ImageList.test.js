import React from "react";
import App from "./App.js";
import renderer from "react-test-renderer";
import ImageList from './ImageList.js';

test('renders ImageList.js correctly', () => {
    const tree = renderer
      .create(<ImageList />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });