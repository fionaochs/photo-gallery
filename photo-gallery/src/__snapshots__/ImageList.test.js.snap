// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageList.js correctly 1`] = `
<section
  className="options"
>
  <section
    className="list-section"
  >
    <ul
      className="creatures"
    />
  </section>
</section>
`;
